Marlin/src/module/stepper.cpp

```diff
@@ -2201,12 +2201,10 @@ hal_timer_t Stepper::calc_timer_interval(uint32_t step_rate) {
   #ifdef CPU_32_BIT
 
     // A fast processor can just do integer division
-    constexpr uint32_t min_step_rate = uint32_t(STEPPER_TIMER_RATE) / HAL_TIMER_TYPE_MAX;
-    return step_rate > min_step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : HAL_TIMER_TYPE_MAX;
+    return step_rate > minimal_step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : HAL_TIMER_TYPE_MAX;
```

Comment:
I took another look and the `step_rate <= minimal_step_rate` case is not correct, although it wasn't correct before either.

`STEPPER_TIMER_RATE` is at 100-500Mhz, divided by some 4/8 prescale. So somewhere around 10-200 million.
However, `HAL_TIMER_TYPE_MAX` is usually 0xFFFFFFFF or ~4 billion. So if we're at the minimal_step_rate of 1 (usually) we'll be waiting for HAL_TIMER_TYPE_MAX ticks which at 10-200 million rate will take in the range of 400-20 seconds.

Maybe switch to:
`return step_rate > minimal_step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : MIN(STEPPER_TIMER_RATE, HAL_TIMER_TYPE_MAX);`

Reply:
The choice to use `HAL_TIMER_TYPE_MAX` to prevent a divide by zero in `calc_timer_interval` originally comes from #25557. The aim was to use the value closest to infinity when the `step_rate` was equal to zero.
```cpp
return step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : HAL_TIMER_TYPE_MAX;
```
This was updated in #25696 to not just apply to a `step_rate` of zero, but to any step rate below a minimum threshold.
```cpp
constexpr uint32_t min_step_rate = uint32_t(STEPPER_TIMER_RATE) / HAL_TIMER_TYPE_MAX;
return step_rate > min_step_rate ? uint32_t(STEPPER_TIMER_RATE) / step_rate : HAL_TIMER_TYPE_MAX;
```
If the value returned from `calc_timer_interval` is later used in a divide by `STEPPER_TIMER_RATE` operation that expects a result >= 1, it makes sense to make that our substitute for infinity, but with this value being 1/8 of `HAL_TIMER_TYPE_MAX`, it means more likelihood of a divide result of 1 or greater.

There is some code that expects `HAL_TIMER_TYPE_MAX` to be returned. Specifically, code that looks for `LA_ADV_NEVER` expects it to be equal to `HAL_TIMER_TYPE_MAX`, and when this value is returned it essentially represents "never."

So, can this really be done? You and me and @tombrazier will need to look closer at this question.